### PR TITLE
PCHR-2543: Override CiviCRM styles for Bootstrap migration

### PR DESCRIPTION
**Pull request by reneolivo**
This changes override CiviCRM styles when wrapping templates inside a #bootstrap-theme tag. The mix between Bootstrap and CiviCRM make some components such as input, and selects to display in an undesired state.

## Before
screenshot 2017-09-04 10 59 07
Button icon position is shifted

screenshot 2017-09-04 10 59 26
Order buttons are smaller than non-bootrstrap version.

screenshot 2017-09-04 10 59 45
* Radio button is broken.
* All inputs have 230px even the ones that are supposed to look smaller.
* Select's dropdown arrow looks broken.

screenshot 2017-09-04 11 04 17
Buttons have wrong border and icons

## After
screenshot 2017-09-04 11 02 55

screenshot 2017-09-04 11 03 08

screenshot 2017-09-04 11 03 27

screenshot 2017-09-04 11 03 52

## Technical Details

* Changes where added to the bootstrap's civiCRM reset file ([source](https://github.com/compucorp/org.civicrm.shoreditch/pull/31/files#diff-2feb88fb3bd71210849d346dcd689f2d)) since it looks like the most appropiate place to fix issues that arise when adding #bootstrap-theme.
* The space between selectors and statements such as `@include`, `@extend` and child selector was removed to match this and other files code structure.
* The `!important` was removed from inputs as it was not needed and was causing all inputs to have a fixed size of 230px even if the input needed to be bigger, smaller, or responsive. ([source](https://github.com/compucorp/org.civicrm.shoreditch/pull/31/files#diff-717f96292f9fd95fa7164dfbced0bd66R20)).

This change does not seem to have broken anything on Admin, and this doesn't affect SSP at all.

### CODE REVIEW

File: base/scss/overrides/reset/_civicrm.scss [hunk 1]
```diff
@@ -50,3 +50,62 @@ table {
...
+    width: 16px;
+    height: 16px;
+  }
+
+  &::after {
+    top: 8px;
+  }
+}
+
+// removes extra padding on select2 ul element added by bootstrap:
```
Comment (igorpavlov-zz) [inline base/scss/overrides/reset/_civicrm.scss:104]: Could you please try the following:
1) Temporary remove these styles that fix the selector
2) Apply a class "no-select2" to the broken selector
What happens if you do it?

**Review by igorpavlov-zz — CHANGES_REQUESTED** [review]

**Comment by igorpavlov** [comment]
Please change the base branch to `staging` if your tests pass.

**Review by igorpavlov-zz — APPROVED** [review]

**Comment by igorpavlov-zz** [comment]
Please change the base branch to `staging` and merge to `staging`.

**Comment by igorpavlov-zz** [comment]
Approved.

**Comment by jamienovick** [comment]
@reneolivo small point - but is that the correct icon for "add" leave and absence type? I think the icon should be a + or something? I would guess that isn't due to this ticket...

**Comment by reneolivo** [comment]
@jamienovick I find that icon odd as well, but it's the one that was being used before these changes. 
screenshot 2017-09-05 11 17 22

The whole button is generated by CiviCRM, but I can try to hide it or even replace it using CSS. Should I do that?

**Comment by reneolivo** [comment]
@jamienovick I have made the change:
screenshot 2017-09-05 11 40 09

Not sure what icon library was being used before, but it seems it's not working very well. The ^ symbol is the default icon, so this must be broken.

To fix it I added the following code:

```scss
// changes the add icon from ^ to +:
.icon.add-icon {
  background: none;

  &::before {
    content: '\f067';
    display: block;
    font-family: FontAwesome;
    text-indent: 0;
  }
}
```

`\f067` is the + symbol in font-awesome. This will change all add icons that are inside #bootstrap-theme so it won't change (or break) CiviCRM as a whole, but at least it will fix the pages that are being migrated to Bootstrap. Is that ok?

**Comment by jamienovick** [comment]
Sounds good to me. So new bootstrap pages will work. I cant comment on the technical but outcome sounds good. thanks!!

**Pull request merged**